teps/0067-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,496 @@
+- Create a standard around how to refer to Tasks which may not may be part of catalog but are used by Pipelines in the catalog
+
+### Non-Goals
+
+- Infrastructure related to the catalog (test infrastructure, …)
+
+- Tools related to catalog maintenance (tekdoc, linter, …)
+
+- Tool to install pipeline and all associated resources
+
+- A tool to display deprecation warnings if a dependent task is deprecated.
+
+### Use Cases (optional)
+
+- Having a pipeline which builds, pushes and deploys an application for a particular language.
+
+- As a CI which can be used to run the test cases, lint checks
+
+- Packaging related CD activities for a particular language, e.g. for publishing a go binary could include testing, linting, building
+
+## Requirements
+
+### Quality
+
+1. Resources should be well documented:
+
+   1. All configuration options are documented
+
+   2. Examples are provided
+
+2. If a resource is in the catalog, a user should feel confident that
+   it will work as advertised See [Tekton Catalog Test Infrastructure for
+   this](https://docs.google.com/document/d/1-czjvjfpuIqYKsfkvZ5RxIbtoFNLTEtOxaZB71Aehdg/edit#)
+
+3. When pipelines have parameters and/or results, it must be
+   clear what values are allowed and the user must receive feedback
+   when they provide values that are not valid
+
+## Proposal
+
+### Support Tiers
+
+This proposal includes [three support
+levels](https://github.com/tektoncd/catalog/issues/5):
+
+1. **Community**
+2. **Verified**
+3. **Official**
+
+Those support tiers are mainly aimed for the Tekton Hub.
+
+For more detailed information about Support Tiers please see [here](https://github.com/tektoncd/community/blob/main/teps/0003-tekton-catalog-organization.md#support-tiers)
+
+### Versioning Resources
+
+Resources definition evolves across time. A new version of the image used is published,
+a new feature in tektoncd/pipeline is available, or a new feature / behaviour is added.
+On the other hand, the catalog should guarantee as much stability as possible for its user.
+
+A version is an identifiable information of a resource along with its kind (Task, Pipeline)
+and name, and it should be present in the definition of the resource. The location of a resource
+in the catalog should be computable given its Kind, Name and the Version.
+e.g. `/{kind}/{name}/{version}/{name}.yaml` - `pipeline/<pipeline-name>/0.5/<pipeline-name>.yaml`
+
+Adding a new version of a resource should be considered as a release of that resource. Here are some cases where a version bump for pipeline may be required
+
+- Task used in the pipeline has been deprecated/removed
+- Location of the task used in the pipeline is changed
+- If anything in the pipeline is changed including params, results, when
+  expressions (which might change the behavior of the pipeline)
+
+#### Questions
+
+- What should be the state of the pipeline when if there's a new version of the task ?
```

Comment:
thanks @PuneetPunamiya !!

Reply:
Thanks for the review and feedback @bobcatfish 🙂